utilities/cli/container.py

```
@@ -772,75 +793,128 @@ def get_environment_args(self) -> List[str]:
+        if not self.dryrun and self._peer_creds_sufficient(display):
+            return []
+
+        if not shutil.which("xauth"):
+            warn(
+                "xauth not found on host; install xauth (or x11-xauth) so X11 "
+                "applications can authenticate inside the container."
+            )
+            return []
+
+        if self.dryrun:
+            placeholder = "/tmp/.docker.xauth"
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/.docker.xauth"

(S108)